### PR TITLE
{bio} [goolf-1.4.10] PRINSEQ-0.20.4-goolf-1.4.10-Perl-5.16.3.eb (REVIEW) 

### PR DESCRIPTION
**Pull request by pescobar**
This application is the typical bioinformatics perl script which depends on many extra perl libraries. How I was managing this kind of installation before was adding the perl libraries dependencies to my main Perl easyconfig. This was leading to a really huge list of libraries / extensions in my main perl easyconfig. Inspired by the bundle easyblock I tried this weird easyconfig which uses the tarball easyblock with extra perl extensions to try to keep the perl script and it's dependencies more self-contained. 

I am curious to listen @boegel complains :P   Do you have any suggestion about a better approach?

I think this can be useful for people deploying bioinfo stuff where this kind of perl applications are quite common CC: @fgeorgatos @pforai @cfenoy @rjeschmi @georg-rath

### CODE REVIEW

**Comment by rjeschmi** [comment]
It looks OK to me. I'll try to test it.

**Comment by boegel** [comment]
@pescobar: this approach makes perfect sense to me

@wpoely86: style review?

File: easybuild/easyconfigs/p/PRINSEQ/PRINSEQ-0.20.4-goolf-1.4.10-Perl-5.16.3.eb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+# sciCORE - University of Basel
+# SIB Swiss Institute of Bioinformatics 
+
+easyblock = 'Tarball'
+
+name = 'PRINSEQ'
+version = '0.20.4'
+
+homepage = 'http://prinseq.sourceforge.net'
+description = """ A bioinformatics tool to PRe-process and show INformation of SEQuence data. """
```
Comment (wpoely86) [inline easybuild/easyconfigs/p/PRINSEQ/PRINSEQ-0.20.4-goolf-1.4.10-Perl-5.16.3.eb]: drop whitespace before and after

**Comment by rjeschmi** [comment]
I'd prefer if it got merged with a newer compiler option too (1.7.20 or a foss option).

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/40022841b8a807d3c5fa for a full test report.

File: easybuild/easyconfigs/p/PRINSEQ/PRINSEQ-0.20.4-goolf-1.4.10-Perl-5.16.3.eb [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    }),
+]
+
+modextrapaths = {
+    'PATH': "", # add installation dir to PATH
+    'PERL5LIB': 'lib/perl5/site_perl/%s/' % (perlver)
+}
+
+postinstallcmds = ["sed -i -e 's|/usr/bin/perl|/usr/bin/env\ perl|' %(installdir)s/*.pl", # fix shebang line
+                   "chmod +x %(installdir)s/*.pl"]  # add execution permission
```
Comment (boegel) [inline easybuild/easyconfigs/p/PRINSEQ/PRINSEQ-0.20.4-goolf-1.4.10-Perl-5.16.3.eb]: please format like:

```python
postinstallcmds = [
    "sed ...",  # fix shebang line
    "chmod ...",  # add execution permisson
]
```

Comment (pforai) [inline easybuild/easyconfigs/p/PRINSEQ/PRINSEQ-0.20.4-goolf-1.4.10-Perl-5.16.3.eb]: This actually needs to be moved to the generic Perl extension block. The Python one implements patching wrong shebangs.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/044ab79951ad2418ac3b for a full test report.

File: easybuild/easyconfigs/p/PRINSEQ/PRINSEQ-0.20.4-goolf-1.4.10-Perl-5.16.3.eb (hunk 2 above)
Comment (boegel) [inline easybuild/easyconfigs/p/PRINSEQ/PRINSEQ-0.20.4-goolf-1.4.10-Perl-5.16.3.eb]: Please open an issue on that in the easyblocks repo?

Or submit a PR for it. ;-)

**Comment by boegel** [comment]
Good to go (moving the `sed` thing to the Perl easyblock will be done later), thanks @pescobar!

**Pull request merged**